imap_processing/tests/idex/test_idex_l2a.py

```diff
@@ -214,14 +222,19 @@ def test_analyze_peaks_warning(caplog):
     tof = np.ones_like(time)
     mass_scale = np.ones_like(time)
     with caplog.at_level("WARNING"):
-        fit_params, area_under_curve = analyze_peaks(tof, time, mass_scale, 0, peaks)
+        fit_params, area_under_curve, chisqr, redchi = analyze_peaks(
+            tof, time, mass_scale, 0, peaks
+        )
     assert any(
         "Failed to fit EMG curve" in message for message in caplog.text.splitlines()
     )
 
     # The fit_params and area_under_curve arrays should be zero
     assert np.all(fit_params == 0)
     assert np.all(area_under_curve == 0)
+    # chi-square and reduced chi-square values should all be np.nan
+    assert np.all(np.isnan(chisqr))
```

Comment:
I think you can use the numpy testing functions here to find out which values aren't equal upon error as well.
`np.testing.assert_array_equal(chisqr, np.nan)`